astropy/modeling/functional_models.py

```diff
@@ -2062,3 +2078,87 @@ def evaluate(cls, x, y, amplitude, r_eff, n, x_0, y_0, ellip, theta):
         z = np.sqrt((x_maj / a) ** 2 + (x_min / b) ** 2)
 
         return amplitude * np.exp(-bn * (z ** (1 / n) - 1))
+
+
+class BlackBody1D(Fittable1DModel):
+    """
+    One dimensional blackbody model.
+
+    Parameters
+    ----------
+    temperature : `~astropy.units.Quantity`
```

Comment:
I am unable to initialize blackbody with a non-quantity, like other models like Gaussian. I propose that we allow using non-Quantity but just assume some default unit.

I also cannot pass in other temperature units that we support; i.e., Celsius and Fahrenheit.